### PR TITLE
Akb 77 create merch details component

### LINKED ISSUE #77: feat f2(2): create merch-details component
1. create new component merch-details that will be used to  show selected product on the page with detail description along with an Add to cart option and buy option
2. create function on click of product div in html
3. create function navigateToDetails() in ts that uses router to navigate to march-details page

 Closes #77

### PR DESCRIPTION
**Pull request by Aman-codde**
1. ng g c components/merch-details

### CODE REVIEW

File: src/client/src/app/components/merch/merch.component.ts [hunk 1]
```diff
@@ -25,4 +27,9 @@ export class MerchComponent implements OnInit {
   ngOnInit(): void {
   }
 
+  navigateToDetails() {
+    console.log("clicked");
```
Comment (brycejjacks) [inline src/client/src/app/components/merch/merch.component.ts]: remember to try to leave console logs out of pull request code.

File: src/client/src/app/components/merch/merch.component.ts [hunk 2]
```diff
@@ -14,7 +15,8 @@ import { Merch } from '../../../../../shared/models/merch.model';
 export class MerchComponent implements OnInit {
   merch$: Observable<Merch[]>
   constructor(
-    private store: Store<AppState>
+    private store: Store<AppState>,
+    private router: Router
```
Comment (brycejjacks) [inline src/client/src/app/components/merch/merch.component.ts]:
```suggestion
    private router: Router,
```
*wanted to see how suggestions work but also i might leave a trailing comma on this

File: src/client/src/app/app-routing.module.ts [hunk 3]
```diff
@@ -19,6 +20,7 @@ const routes: Routes = [
   {path: 'games', loadChildren:
   () => import('./modules/games/module-games.module').then(m => m.GamesModule)
   },
+  {path: 'merch-details', component: MerchDetailsComponent},
```
Comment (brycejjacks) [inline src/client/src/app/app-routing.module.ts]: i might also try adding a this in the routes array in alphabetical order, or next to like routes for organization

**Review by brycejjacks — APPROVED** [review]
good job... ive added some comments but i dont want them to be blocking or anything.. maybe ill resolve them immediately or ask vinson about how that goes ??

File: src/client/src/app/components/merch/merch.component.ts (hunk 1 above)
Comment (Aman-codde) [reply src/client/src/app/components/merch/merch.component.ts]: good catch! I ll remove that

**Review by ChelleAmon — APPROVED** [review]
Not so much logic that needs to be checked. Great job! I think you need to click on 'resolve conversation' so we can merge this PR.

File: src/client/src/app/components/merch/merch.component.ts (hunk 1 above)
Comment (brycejjacks) [reply src/client/src/app/components/merch/merch.component.ts]: ok and i guess you would resolve this after your change it??

**Pull request merged**